open-rmf/rmf-web
# WIP taskv2 - dashboard

## Self-checks

- [x] I have prototyped this new feature (if necessary) on Figma 
- [x] I tested the behavior of the components that interact with the backend, with an e2e test
- [x] Task progress paper looks funny.
- [x] coloring of task logs containers (don't know according to what yet)
- [x] scroll bar in logs and progress
- [x] color task log boxes (yellow log: warnings, delays, etc
red log: errors, manual interventions required, failures, etc)
- [x] Using event name and phase category from the `taskState` in the `taskLog` panel (this info is not part of the log, but the state)
- [x] test the tree with two level dependency events
- [x] Move Timeline to react components  
- [ ] Override TimeLineDot default colors using by the theme colors (Don't know how to do this yet if someone has an idea, it seems a very minor thing at the moment so I'm not going to invest more time into it)
- [ ] remove duplicates events in the tree (should appear once in case of dependency) 
- [ ] scale down events in the tree as it goes down
- [ ] I tried testing edge cases
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components

## Discussion
To test:
`ros2 launch rmf_demos_ign office.launch.xml`
for api server
`npm run start`
for dashboard
`npm run start:react`
for task
`ros2 launch rmf_demos office_conflict.launch.xml`

Then you should see something like this

![Screenshot from 2022-01-28 00-09-20](https://user-images.githubusercontent.com/10261903/151444093-33695eec-c5e1-4315-b156-fc6455033faa.png)
